File: scripts/generate-icons.py
#!/usr/bin/env python3
import sys
from PIL import Image
import os

try:
    # Load mascot image
    mascot = Image.open('/vercel/share/v0-project/public/mascot.jpg')
    
    # Convert to RGBA if needed
    if mascot.mode != 'RGBA':
        mascot = mascot.convert('RGBA')
    
    # Create light version (32x32)
    light_icon = mascot.resize((32, 32), Image.Resampling.LANCZOS)
    light_icon.save('/vercel/share/v0-project/public/icon-light-32x32.png')
    print("[v0] Created icon-light-32x32.png")
    
    # Create dark version (32x32) - same as light for now
    dark_icon = mascot.resize((32, 32), Image.Resampling.LANCZOS)
    dark_icon.save('/vercel/share/v0-project/public/icon-dark-32x32.png')
    print("[v0] Created icon-dark-32x32.png")
    
    # Create standard icon (192x192)
    standard_icon = mascot.resize((192, 192), Image.Resampling.LANCZOS)
    standard_icon.save('/vercel/share/v0-project/public/icon.png')
    print("[v0] Created icon.png")
    
    # Create apple icon (180x180)
    apple_icon = mascot.resize((180, 180), Image.Resampling.LANCZOS)
    apple_icon.save('/vercel/share/v0-project/public/apple-icon.png')
    print("[v0] Created apple-icon.png")
    
    print("[v0] All icons generated successfully!")
    
except Exception as e:
    print(f"[v0] Error: {e}", file=sys.stderr)
    sys.exit(1)

File: scripts/generate-icons.mjs
#!/usr/bin/env node

import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const mascotPath = './public/mascot.jpg';
const outputDir = './public';

async function generateIcons() {
  console.log('[v0] Starting icon generation from mascot...');
  
  try {
    // Generate 32x32 light version
    await sharp(mascotPath)
      .resize(32, 32, { fit: 'cover' })
      .png()
      .toFile(path.join(outputDir, 'icon-light-32x32.png'));
    console.log('[v0] Generated icon-light-32x32.png');

    // Generate 32x32 dark version (same image, just renamed)
    await sharp(mascotPath)
      .resize(32, 32, { fit: 'cover' })
      .png()
      .toFile(path.join(outputDir, 'icon-dark-32x32.png'));
    console.log('[v0] Generated icon-dark-32x32.png');

    // Generate SVG version (convert PNG to SVG-like output, but we'll use PNG scaled up)
    await sharp(mascotPath)
      .resize(256, 256, { fit: 'cover' })
      .png()
      .toFile(path.join(outputDir, 'icon.png'));
    console.log('[v0] Generated icon.png');

    // For Apple icon
    await sharp(mascotPath)
      .resize(180, 180, { fit: 'cover' })
      .png()
      .toFile(path.join(outputDir, 'apple-icon.png'));
    console.log('[v0] Generated apple-icon.png');

    console.log('[v0] Icon generation completed successfully!');
  } catch (error) {
    console.error('[v0] Error generating icons:', error);
    process.exit(1);
  }
}

generateIcons();
